pandas/tests/scalar/period/test_period.py

```diff
@@ -509,35 +235,253 @@ def test_period_constructor_offsets(self):
+        exp = Period('1989Q3', freq=freq)
+        assert '1989Q3' in str(exp)
+        stamp = exp.to_timestamp('D', how='end')
+        p = Period(stamp, freq=freq)
+        assert p == exp
+
+        stamp = exp.to_timestamp('3D', how='end')
+        p = Period(stamp, freq=freq)
+        assert p == exp
+
+    @pytest.mark.parametrize('month', MONTHS)
+    def test_period_cons_annual(self, month):
+        # bugs in scikits.timeseries
+        freq = 'A-%s' % month
+        exp = Period('1989', freq=freq)
+        stamp = exp.to_timestamp('D', how='end') + timedelta(days=30)
+        p = Period(stamp, freq=freq)
+
+        with tm.assert_produces_warning(FutureWarning):
+            assert p == exp + 1
+        assert isinstance(p, Period)
+
+    @pytest.mark.parametrize('day', DAYS)
+    @pytest.mark.parametrize('num', range(10, 17))
+    def test_period_cons_weekly(self, num, day):
+        daystr = '2011-02-%d' % num
+        freq = 'W-%s' % day
+
+        result = Period(daystr, freq=freq)
+        expected = Period(daystr, freq='D').asfreq(freq)
+        assert result == expected
+        assert isinstance(result, Period)
+
+    def test_period_from_ordinal(self):
+        p = Period('2011-01', freq='M')
+        res = Period._from_ordinal(p.ordinal, freq='M')
+        assert p == res
+        assert isinstance(res, Period)
+
+    def test_period_cons_nat(self):
+        p = Period('NaT', freq='M')
+        assert p is NaT
+
+        p = Period('nat', freq='W-SUN')
+        assert p is NaT
+
+        p = Period(iNaT, freq='D')
+        assert p is NaT
+
+        p = Period(iNaT, freq='3D')
+        assert p is NaT
+
+        p = Period(iNaT, freq='1D1H')
+        assert p is NaT
+
+        p = Period('NaT')
+        assert p is NaT
+
+        p = Period(iNaT)
+        assert p is NaT
+
+    def test_period_cons_mult(self):
+        p1 = Period('2011-01', freq='3M')
+        p2 = Period('2011-01', freq='M')
+        assert p1.ordinal == p2.ordinal
+
+        assert p1.freq == offsets.MonthEnd(3)
+        assert p1.freqstr == '3M'
+
+        assert p2.freq == offsets.MonthEnd()
+        assert p2.freqstr == 'M'
+
+        with tm.assert_produces_warning(FutureWarning):
+            result = p1 + 1
+            assert result.ordinal == (p2 + 3).ordinal
+
+        assert result.freq == p1.freq
+        assert result.freqstr == '3M'
+
+        with tm.assert_produces_warning(FutureWarning):
+            result = p1 - 1
+            assert result.ordinal == (p2 - 3).ordinal
+        assert result.freq == p1.freq
+        assert result.freqstr == '3M'
+
+        msg = ('Frequency must be positive, because it'
+               ' represents span: -3M')
+        with pytest.raises(ValueError, match=msg):
+            Period('2011-01', freq='-3M')
+
+        msg = ('Frequency must be positive, because it' ' represents span: 0M')
+        with pytest.raises(ValueError, match=msg):
+            Period('2011-01', freq='0M')
+
```

Comment:
can parametrize next pass (and maybe split to 2 tests, working and error condtiions)

Reply:
Yah, I've got another branch focusing more closely on the scalar tests, will work this in there.